File: solutions/day_01.py
""" day_01.py -- Advent of Code 2020 Day 1: Report Repair

    Author: Chris Bowman
    Last Modified: 12/2/2020
    License: MIT
"""


from utils import integers

GOAL = 2020


def part_1(data: list) -> int:
    for a in data:
        for b in data[data.index(a):]:
            if a + b == GOAL:
                return a * b


def part_2(data: list) -> int:
    for a in data:
        for b in data[data.index(a):]:
            for c in data[data.index(b):]:
                if a + b + c == GOAL:
                    return a * b * c


def main():
    d = integers(open('01').read())
    print(part_1(d))
    print(part_2(d))


if __name__ == '__main__':
    main()

File: solutions/day_01.jl

# Advent of Code 2020 Day 1

function part_1()
    lines = readlines(open("../inputs/01"))
    GOAL = 2020
    d = map(x->parse(Int64,x), lines)
    for a in d
        for b in d
            if a != b && a + b == GOAL
                return a * b
                break
            end
        end
    end
end


function part_2()
    lines = readlines(open("../inputs/01"))
    GOAL = 2020
    d = map(x->parse(Int64,x), lines)
    for a in d
        for b in d
            for c in d
                if a != b && b != c && a + b + c == GOAL
                    return a * b * c
                end
            end
        end
    end
end

print(part_1())
print(part_2())